cpp/oneapi/dal/algo/triangle_counting/test/triangle_counting_test.cpp

```diff
@@ -0,0 +1,394 @@
+                                          3, 4, 5, 6, 7, 8, 0, 1, 2, 4, 5, 6, 7, 8, 0, 1, 2, 3,
+                                          5, 6, 7, 8, 0, 1, 2, 3, 4, 6, 7, 8, 0, 1, 2, 3, 4, 5,
+                                          7, 8, 0, 1, 2, 3, 4, 5, 6, 8, 0, 1, 2, 3, 4, 5, 6, 7 };
+    std::array<std::int64_t, 10> rows = { 0, 8, 16, 24, 32, 40, 48, 56, 64, 72 };
+    std::array<std::int64_t, 9> local_triangles = { 28, 28, 28, 28, 28, 28, 28, 28, 28 };
+};
+
+class acyclic_graph_8_type : public graph_base_data {
+public:
+    acyclic_graph_8_type() {
+        vertex_count = 8;
+        edge_count = 7;
+        global_triangle_count = 0;
+    }
+
+    std::array<std::int32_t, 8> degrees = { 3, 1, 3, 3, 1, 1, 1, 1 };
+    std::array<std::int32_t, 14> cols = { 1, 2, 4, 0, 0, 3, 6, 2, 5, 7, 0, 3, 2, 3 };
+    std::array<std::int64_t, 9> rows = { 0, 3, 4, 7, 10, 11, 12, 13, 14 };
+    std::array<std::int64_t, 8> local_triangles = { 0, 0, 0, 0, 0, 0, 0, 0 };
+};
+
+class two_vertices_graph_type : public graph_base_data {
+public:
+    two_vertices_graph_type() {
+        vertex_count = 2;
+        edge_count = 1;
+        global_triangle_count = 0;
+    }
+
+    std::array<std::int32_t, 2> degrees = { 1, 1 };
+    std::array<std::int32_t, 2> cols = { 1, 0 };
+    std::array<std::int64_t, 3> rows = { 0, 1, 2 };
+    std::array<std::int64_t, 2> local_triangles = { 0, 0 };
+};
+
+class cycle_graph_9_type : public graph_base_data {
+public:
+    cycle_graph_9_type() {
+        vertex_count = 9;
+        edge_count = 9;
+        global_triangle_count = 0;
+    }
+
+    std::array<std::int32_t, 9> degrees = { 2, 2, 2, 2, 2, 2, 2, 2, 2 };
+    std::array<std::int32_t, 18> cols = { 1, 8, 0, 2, 1, 3, 2, 4, 3, 5, 4, 6, 5, 7, 6, 8, 0, 7 };
+    std::array<std::int64_t, 10> rows = { 0, 2, 4, 6, 8, 10, 12, 14, 16, 18 };
+    std::array<std::int64_t, 9> local_triangles = { 0, 0, 0, 0, 0, 0, 0, 0, 0 };
+};
+
+class triangle_graph_type : public graph_base_data {
+public:
+    triangle_graph_type() {
+        vertex_count = 3;
+        edge_count = 3;
+        global_triangle_count = 1;
+    }
+
+    std::array<std::int32_t, 3> degrees = { 2, 2, 2 };
+    std::array<std::int32_t, 6> cols = { 1, 2, 0, 2, 0, 1 };
+    std::array<std::int64_t, 4> rows = { 0, 2, 4, 6 };
+    std::array<std::int64_t, 3> local_triangles = { 1, 1, 1 };
+};
+
+class wheel_graph_6_type : public graph_base_data {
+public:
+    wheel_graph_6_type() {
+        vertex_count = 6;
+        edge_count = 10;
+        global_triangle_count = 5;
+    }
+
+    std::array<std::int32_t, 6> degrees = { 5, 3, 3, 3, 3, 3 };
+    std::array<std::int32_t, 20> cols = {
+        1, 2, 3, 4, 5, 0, 2, 5, 0, 1, 3, 0, 2, 4, 0, 3, 5, 0, 1, 4
+    };
+    std::array<std::int64_t, 7> rows = { 0, 5, 8, 11, 14, 17, 20 };
+    std::array<std::int64_t, 6> local_triangles = { 5, 2, 2, 2, 2, 2 };
+};
+
+class isolated_vertices_graph_10_type : public graph_base_data {
```

Comment:
It is not isolated according to the rows and cols fields.

Reply:
What graph do you call "isolated vertices"?

Reply:
Graph that contains isolated vertices #3, #6 and #8

Reply:
Do you mean that it makes sense to rename graph to "graph_with_isolated_vertices_10_type"?

Reply:
Yes.